### PR TITLE
C++: Remove `safeFloor` in simple range analysis

### PR DESCRIPTION
**Pull request by paldepind**
This PR removes the `safeFloor` predicate from the simple range analysis library. The predicate is a workaround for the fact that QL's `floor` operates on 32 bit integers. We no longer need this after the recently introduced `floorFloat` predicate, which operates directly on `float` without any loss in range.

In addition to getting rid of `safeFloor`, as the test shows, this also enables correct flooring on fractional numbers greater than 2^32. This problem was pointed out by @geoffw0 back in https://github.com/github/codeql/pull/3445:
> There are some edge cases for int64 types that we don't handle well (because QL floor() fails on large numbers!) so such cases are simply excluded from the analysis.

### CODE REVIEW

File: cpp/ql/test/library-tests/rangeanalysis/SimpleRangeAnalysis/test.c [hunk 1]
```diff
@@ -333,6 +333,16 @@ int test_mult05(int a, int b) {
   return total;
 }
 
+// Tests for shift operators.
+unsigned long long test_shift(unsigned long long a) {
+  // `odd` is the largest odd integer that can be represented by a double.
```
Comment (geoffw0) [inline cpp/ql/test/library-tests/rangeanalysis/SimpleRangeAnalysis/test.c]:
```suggestion
  // `odd` is the largest odd integer that can be precisely represented by a double.
```

**Review by geoffw0 — DISMISSED** [review]
LGTM.  Glad to see the back of `safeFloor`.  :+1:

Please check there are no surprises when the DCA run finishes before merging.

**Comment by paldepind** [comment]
> Please check there are no surprises when the DCA run finishes before merging.

The DCA report seems fine, it shows something about a bad join, but I don't think it's related to the PR?
